newapi/ooniapi/measurements.py

```diff
@@ -184,32 +186,39 @@ def _fetch_jsonl_measurement_body_inner(
         if n == linenum:
             return line
 
-    return b""
+    raise MsmtNotFound
 
 
-def _fetch_jsonl_measurement_body(report_id, input: str) -> bytes:
+def _fetch_jsonl_measurement_body(report_id, input: str, measurement_uid) -> bytes:
     """Fetch jsonl from S3, decompress it, extract msmt"""
-    query = """SELECT s3path, linenum
-    FROM jsonl
-    WHERE report_id = :report_id
-    AND input = :input
-    """
-    if input is None:
-        input = ""
-    query_params = dict(input=input, report_id=report_id)
+    query = "SELECT s3path, linenum FROM jsonl "
+    if measurement_uid is None:
+        query += "WHERE report_id = :report_id AND input = :input LIMIT 1"
+        query_params = dict(input=input, report_id=report_id)
+
+    else:
+        query += """WHERE measurement_uid = :mid
+        OR (report_id = :rid AND input = :inp)
```

Comment:
Why do we need the OR here? If the `measurement_uid` has been passed, shouldn't it always return a match?

Reply:
No, we don't have uids for most msmts

Reply:
Even after reprocessing all of the legacy ones?

Reply:
Yes, they are the ones with uid. The new ones don't have it.

Reply:
Ah right, so until we reprocess also the new data, we won't have the measurement_uid in the jsonl table, but only inside of the fastpath table. Is this correct?

Reply:
Yes. After we reprocess everything we'll be able to simplify the query, also drop the autoclaved-related queries, and remove a DB index.